src/utils/async.ts

```diff
@@ -49,3 +49,24 @@ export const waitFor = async (
     }, interval);
   });
 };
+
+let timer: NodeJS.Timeout | null = null;
+let immediateCall = true;
+let lastCallTime = 0;
+
+export const debounceFunction = async (func: () => Promise<void>) => {
```

Comment:
Is there a reason why we can't use the `debounce` fuction from `lodash` instead of writing our own? My concern would be the shared variables here being used by simultaneous calls to `debounceFunction`.

Reply:
This has been updated to use `lodash.throttle`. Throttle is the right function to use.